src/script/api/script_airport.cpp

```diff
@@ -67,7 +67,8 @@
 {
 	if (!IsAirportInformationAvailable(type)) return -1;
 
-	return _settings_game.station.modified_catchment ? ::AirportSpec::Get(type)->catchment : (uint)CA_UNMODIFIED;
+	const auto coverage = _settings_game.station.modified_catchment ? ::AirportSpec::Get(type)->catchment : (uint)CA_UNMODIFIED;
+	return (type != AT_OILRIG || _settings_game.station.serve_neutral_industries) ? coverage : 0;
```

Comment:
```suggestion
	if (type == AT_OILRIG && !_settings_game.station.serve_neutral_industries) return 0;
	return _settings_game.station.modified_catchment ? ::AirportSpec::Get(type)->catchment : (uint)CA_UNMODIFIED;
```
Why calculate the coverage when you know it's going to be zero anyway?